src/planner/dart/ConfigurationToEndEffectorOffset.cpp

```diff
@@ -48,13 +106,30 @@ ConfigurationToEndEffectorOffset::getEndEffectorBodyNode() const
 const statespace::dart::MetaSkeletonStateSpace::State*
 ConfigurationToEndEffectorOffset::getStartState() const
 {
+  // Take start state from MetaSkeleton if passed. Store in the ScopedState
+  // instance variable to avoid dangling pointers.
+  if (mMetaSkeleton)
+  {
+    auto startState = mMetaSkeletonStateSpace->createState();
+    mMetaSkeletonStateSpace->getState(mMetaSkeleton.get(), startState);
+    mStateSpace->copyState(startState, mStartState);
+  }
+
   return mStartState;
 }
 
 //==============================================================================
-const Eigen::Vector3d& ConfigurationToEndEffectorOffset::getDirection() const
+const Eigen::Vector3d ConfigurationToEndEffectorOffset::getDirection() const
 {
-  return mDirection;
+  if (!mDirection)
+  {
+    return mEndEffectorBodyNode->getWorldTransform()
```

Comment:
Seems like this is copied from `ConcreteManipulator::getEndEffectorDirection`, maybe we can try to reuse that instead?

Reply:
After thinking about it, just moving this method into robot/util seems like the best way around this.